pkg/volume/gce_pd/gce_util.go

```diff
@@ -122,6 +122,167 @@ func (util *GCEDiskUtil) DetachDisk(pd *gcePersistentDisk) error {
 	if err := gce.(*gce_cloud.GCECloud).DetachDisk(pd.pdName); err != nil {
 		return err
 	}
+
+	// Verify disk detached, retry if needed.
+	go verifyDiskDetachedRetryIfNeeded(pd, gce)
+	return nil
+}
+
+// Verifys the disk device to be created has been succesffully attached, and retries if it fails.
+func verifyDiskAttachedRetryIfNeeded(pd *gcePersistentDisk, sdBeforeSet util.StringSet, gce cloudprovider.Interface) (string, error) {
```

Comment:
nit: ensureAttached() or verifyAttached() rather?  Disk and RetryIfNeeded are implied..